### PR TITLE
Loginpage

### PR DESCRIPTION
**Pull request by vishalg8454**
This PR adds the LoginPage into E-commerce app. A guest user has been provided.
The components added are-

- `Toast` at src/components/Toast.jsx
 
The routes added are

- `LoginPage` at ("/login") at src/routes/loginpage/LoginPage.jsx

The context added is

- `toast-context` at src/context/toast-context.js
- `user-context` at src/context/user-context.js

[live preview](https://deploy-preview-3--bookstock.netlify.app/)
Would appreciate you feedback and suggestions.

image

image

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -0,0 +1 @@
+REACT_APP_JWT_SECRET=something
```
Comment (viru2001) [inline .env:1]: you should not push your .env file on github

File: src/routes/loginpage/loginpage.css [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+}
+.form-wrapper{
+    margin: 2rem;
+    padding: 2rem;
+    display: flex;
+    flex-direction: column;
+    max-width: 20rem;
+    margin-left: auto;
+    margin-right: auto;
+    box-shadow: 0px 0px 10px 4px rgb(0 0 0 / 5%);
```
Comment (sahiba0915) [inline src/routes/loginpage/loginpage.css:13]: can use color variables

**Review by sahiba0915 — APPROVED** [review]
Just added one suggestion, rest code looks perfect to me and good use of toast component.

File: src/routes/homepage/Homepage.jsx [hunk 3]
```diff
@@ -25,7 +25,7 @@ export const Homepage = () => {
   
   useEffect(() => {
     document.title = "BookStock";
-  });
+  },[]);
```
Comment (viru2001) [inline src/routes/homepage/Homepage.jsx:28]: you can make custom hook for changing document title

**Review by viru2001 — APPROVED** [review]
- code is good and easily understandable.
- variable names are proper

gave small suggestion on which you could work on

File: src/routes/loginpage/loginpage.css (hunk 2 above)
Comment (vishalg8454) [reply src/routes/loginpage/loginpage.css:13]: sure, will take care.

File: .env (hunk 1 above)
Comment (vishalg8454) [reply .env:1]: Ah yes, this was a workaround. I will learn how to use .env with Netlify and implement this in near future.

File: src/routes/homepage/Homepage.jsx (hunk 3 above)
Comment (vishalg8454) [reply src/routes/homepage/Homepage.jsx:28]: sure, good suggestion!

**Pull request merged**